### PR TITLE
Converted specs to RSpec3 syntax with the help of transpec.

### PR DESCRIPTION
**Pull request by juliandunn**
I removed all the pinning on mixlib-config, chef, rspec, etc.

### CODE REVIEW

File: knife-ec2.gemspec [hunk 1]
```diff
@@ -19,10 +19,10 @@ Gem::Specification.new do |s|
...
 
-  s.add_development_dependency 'mixlib-config', '~> 2.0'
-  s.add_development_dependency 'chef',          '>= 11.16.2', '< 12'
-  s.add_development_dependency 'rspec',         '~> 2.14'
-  s.add_development_dependency 'rake',          '~> 10.1'
+  s.add_development_dependency 'mixlib-config'
+  s.add_development_dependency 'chef'
+  s.add_development_dependency 'rspec'
+  s.add_development_dependency 'rake'
   s.add_development_dependency 'sdoc',          '~> 0.3'
```
Comment (adamleff) [inline knife-ec2.gemspec]: Per our discussion, it's probably best to still pin these, at least adhering to what we support in chef-dk.  I matched the versions to the 0.7.0 RC of chef-dk as follows:

```ruby
  s.add_development_dependency 'chef',  '~> 12.0', '>= 12.2.1'
  s.add_development_dependency 'rspec', '~> 3.0'
  s.add_development_dependency 'rake',  '~> 10.1'
  s.add_development_dependency 'sdoc',  '~> 0.3'
```

rspec tests pass, and running actual commands against AWS succeeded.  I'd recommend that we adopt these unless there's a reason not to.  What say you, @juliandunn? :smile:

Comment (juliandunn) [inline knife-ec2.gemspec]: I buy it - I'm pushing a change now.

**Comment by adamleff** [comment]
I'm :+1: on this pending clean tests after 94ae8ed.  Per Matt, we should wait for @ryancragun to give this a :+1: too before merging.  Thanks for doing this, @juliandunn!

**Comment by ryancragun** [comment]
:ship:

File: knife-ec2.gemspec (hunk 1 above)
Comment (gbish) [inline knife-ec2.gemspec]: Hi,

Just wanted to say hi and tell you how much I miss you, here are some interesting things for you <http://ntiprorytho.mikesdogshack.info/oitajwc>

Be well, hebgb@blueyonder.co.uk

**Pull request merged**